src/components/icon/icon.ts

```diff
@@ -0,0 +1,209 @@
+ *   "[namespace]:[name]", if not the value will be the name of an icon in the default namespace.
+ *   Examples:
+ *     <md-icon svgIcon="left-arrow"></md-icon>
+ *     <md-icon svgIcon="animals:cat"></md-icon>
+ *
+ * - Use a font ligature as an icon by putting the ligature text in the content of the <md-icon>
+ *   component. By default the Material icons font is used as described at
+ *   http://google.github.io/material-design-icons/#icon-font-for-the-web. You can specify an
+ *   alternate font by setting the fontSet input to either the CSS class to apply to use the
+ *   desired font, or to an alias previously registered with MdIconRegistry.registerFontClassAlias.
+ *   Examples:
+ *     <md-icon>home</md-icon>
+ *     <md-icon fontSet="myfont">sun</md-icon>
+ *
+ * - Specify a font glyph to be included via CSS rules by setting the fontSet input to specify the
+ *   font, and the fontIcon input to specify the icon. Typically the fontIcon will specify a
+ *   CSS class which causes the glyph to be displayed via a :before selector, as in
+ *   https://fortawesome.github.io/Font-Awesome/examples/
+ *   Example:
+ *     <md-icon fontSet="fa" fontIcon="alarm"></md-icon>
+ */
+@Component({
+  template: '<ng-content></ng-content>',
+  selector: 'md-icon',
+  styleUrls: ['./components/icon/icon.css'],
+  host: {
+    'role': 'img',
+  },
+  directives: [NgClass],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdIcon implements OnChanges, OnInit, AfterContentChecked {
+  @Input() svgSrc: string;
+  @Input() svgIcon: string;
+  @Input() fontSet: string;
+  @Input() fontIcon: string;
+  @Input() alt: string;
+
+  @Input('aria-label') ariaLabelFromParent: string = '';
+
+  private _previousFontSetClass: string;
+  private _previousFontIconClass: string;
+
+  constructor(
+      private _element: ElementRef,
+      private _renderer: Renderer,
+      private _changeDetectorRef: ChangeDetectorRef,
+      private _mdIconRegistry: MdIconRegistry) {
+  }
+
+  /**
+   * Splits an svgIcon binding value into its icon set and icon name components.
+   * Returns a 2-element array of [(icon set), (icon name)].
+   * The separator for the two fields is ':'. If there is no separator, an empty
+   * string is returned for the icon set and the entire value is returned for
+   * the icon name. If the argument is falsy, returns an array of two empty strings.
+   * Examples:
+   *   'social:cake' -> ['social', 'cake']
+   *   'penguin' -> ['', 'penguin']
+   *   null -> ['', '']
+   */
+  private _splitIconName(iconName: string): [string, string] {
+    if (!iconName) {
+      return ['', ''];
+    }
+    const separatorIndex = this.svgIcon.indexOf(':');
+    if (separatorIndex == -1) {
+      return ['', iconName];
+    }
+    return [iconName.substring(0, separatorIndex), iconName.substring(separatorIndex + 1)];
```

Comment:
I'm probably missing something, but any reason we're not using `iconName.split(':')`?  (and checking the arr's length to see if ":" was present)